src/trunk.c

```diff
@@ -7047,7 +7088,7 @@ trunk_range(trunk_handle  *spl,
 
 destroy_range_itor:
    trunk_range_iterator_deinit(range_itor);
-   platform_free(spl->heap_id, range_itor);
+   platform_free(PROCESS_PRIVATE_HEAP_ID, range_itor);
```

Comment:
One of the instances where we fall-back to `malloc()` from the heap.

Once part-3 is finalized and integrated, it will be a good opportunity to clean this up. Now that we have free-fragment recycling working in part-3, this kind of bypass may probably be unnecessary. But I'd rather not have to do anything about it right now, till all this work gets integrated.